app/src/main/java/nu/yona/app/ui/LaunchActivity.java

```diff
@@ -141,12 +141,18 @@ private void reloadUserFromServer(String url)
 	{
 		if (!AppUtils.isNetworkAvailable(YonaApplication.getAppContext()))
 		{
-			AppUtils.displayErrorAlert(this, new ErrorMessage(this.getString(R.string.mandatory_user_fetch_failure)));
+			AppUtils.displayErrorAlert(this, new ErrorMessage(this.getString(R.string.mandatory_user_fetch_failure)), this::closeApplicationIfNoNetwork);
 			return;
 		}
 		getUserFromServer(url);
 	}
 
+	public void closeApplicationIfNoNetwork(DialogInterface dialog, int which)
```

Comment:
This name would be right if the method would close the application if there is no network, but it closes it always, irrespective of the network connectivity. So just call it ``closeApplication``.